Sources/SourceKitLSP/Swift/ClosureCompletionFormat.swift

```diff
@@ -0,0 +1,64 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public import SwiftSyntax
+public import SwiftBasicFormat
+
+/// A specialization of `BasicFormat` for closure literals in a code completion
+/// context.
+///
+/// This is more conservative about newline insertion: unless the closure has
+/// multiple statements in its body it will not be reformatted to multiple
+/// lines.
+@_spi(Testing)
+public class ClosureCompletionFormat: BasicFormat {
+  public override func requiresNewline(between first: TokenSyntax?, and second: TokenSyntax?) -> Bool {
```

Comment:
I would add `@_spi(Testing)` here as well, just to be clear that it’s not a properly public declaration.

```suggestion
  @_spi(Testing) public override func requiresNewline(between first: TokenSyntax?, and second: TokenSyntax?) -> Bool {
```

Reply:
Makes sense; done!